### ISSUE TITLE
Audio Uniform

### ISSUE THREAD

**Issue by zz85** [mrdoob/glsl-sandbox#12]
this has been on my (and perhaps others') mind and it would probably take a day to integrate but adding this here so I might get to do it sometime ;)

```
uniform float audio;
```

the idea is to run a spectrum analyzer on some background music and pass in the dynamics as an `audio` uniform of value between 0 and 1.

and i initially thought of using the audio wrapper I wrote https://github.com/zz85/audiokeys.js but perhaps we could use the soundcloud or echonest api (which is probably easier to host it on heroku, while providing music in the sandbox :) too?

**Comment by Sur3** [comment]
Yes i also like the idea of audio filters, we just need an output audio array of size resolution.x*resolution.y and a samplerate related to time, then we could fill that every frame to get a nice samplerate, we could then also generate sounds, e.g. a simple sinewave-based chord:
audio_time=samplerate*time;
audio_pos=gl_FragCoord.x*resolution.y+gl_FragCoord.y+audio_time;
audio[out]=sin(audio_pos*frequency);
audio[out]+=sin(audio_pos*frequency*pow(2.0,3.0/12.0));
audio[out]+=sin(audio_pos*frequency*pow(2.0,7.0/12.0));
